test/test-fs-event.c

```diff
@@ -162,10 +156,7 @@ static void fs_event_create_files(uv_timer_t* handle) {
   if (++fs_event_created < fs_event_file_count) {
     /* Create another file on a different event loop tick.  We do it this way
      * to avoid fs events coalescing into one fs event. */
```

Comment:
We explicitly request coalescing of events quicker than 50ms apart (after the first event is delivered immediately). Maybe we should adjust that?
https://github.com/libuv/libuv/blob/v1.x/src/unix/fsevents.c#L345

Reply:
Or maybe there is something to be done with `FSEventStreamFlush[A]Sync` (which forces libuv to wait for delivery of all current events that were otherwise waiting for the timeout to elapse)?

Reply:
It is also written to discard all events that happened before any uv_fs_event_start call on all existing watchers, which seems like an actual bug here:
https://github.com/libuv/libuv/blob/a3de1384c304553886c4e74dc82d9356df732cf8/src/unix/fsevents.c#L441-L481

This seems like it should be fetching the current eventId at the time of the actual uv_fs_event_start call, and using that to ensure continuity of events on the existing watchers (along with a FSEventStreamFlush[A]Sync call, to ensure we don't pick up old events for the new paths)

Reply:
As I remember it, it's written that way due to FSEvent's delivery of pre-existing events being buggy. Maybe that's fixed now, it's been almost 10 years.

I'll merge this PR and open an issue to investigate further.